BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/single_file/single_file.rb

```diff
@@ -80,4 +80,25 @@ def gzip_content(operation = :compress) #:nodoc:
     self.sync_to_provider
   end
 
+  # compute file md5 hash sum (useful to avoid tampering, e.g. with task archive)
+  def cache_compute_md5sum
+    md5command =
+        case CBRAIN::System_Uname
+        when /Linux/i
+          "md5sum"
+        when /Darwin/i
+          "md5"
+        else
+          "md5sum"
+        end
+    self.sync_to_cache
```

Comment:
Do NOT do the sync to cache. Expect the programmer to have done it already. That's not within the scope of the method.

Reply:
Another reason for this is that a programmer might want to compute the md5 of something that he has prepared in the cache but has not yet been pushed to the provider side. If you do a sync to cache, you destroy the cache.

Reply:
done